robot/client/client.go

```diff
@@ -593,9 +593,13 @@ func (rc *RobotClient) DiscoverComponents(ctx context.Context, qs []discovery.Qu
 
 	discoveries := make([]discovery.Discovery, 0, len(resp.Discovery))
 	for _, disc := range resp.Discovery {
+		m, err := resource.NewModelFromString(disc.Query.Model)
+		if err != nil {
+			return nil, err
+		}
 		q := discovery.Query{
 			SubtypeName: resource.SubtypeName(disc.Query.Subtype),
-			Model:       disc.Query.Model,
+			Model:       m,
```

Comment:
backwards compat q here

Reply:
I don't know enough about how/where this is used in App (elsewhere?) to provide a good answer. It passes tests here in RDK, but I expect changes will be needed on the App side. Also, I think I overlooked that this is using SubtypeName here instead of just "Subtype" so probably need to change that as well (like I did in most other "registry" like places)